lib/wasi/src/macros.rs

```diff
@@ -31,6 +31,6 @@ macro_rules! wasi_try {
 /// corrupted.
 macro_rules! get_input_str {
     ($memory:expr, $data:expr, $len:expr) => {{
-        wasi_try!($data.get_utf8_str($memory, $len), __WASI_EINVAL)
+        wasi_try!($data.get_utf8_string($memory, $len), __WASI_EINVAL)
```

Comment:
Why not keeping `get_utf8_str`? It returns a `Cow`, so you can easily transform it into a `String` if needed.

Reply:
I forgot why, we can try to use `get_utf8_str` but I believe I did this because the other didn't work in both js and sys